GUIDELINES.md

```diff
@@ -0,0 +1,41 @@
+# Instrumentations Implementation Guide
+
+This document captures general guidelines for implementing instrumentations in NodeJS and browser.
```

Comment:
```suggestion
This document captures general guidelines for implementing instrumentations.
```
I assume nothing would change for e.g. deno so why limiting?

Reply:
I was just thinking to clarify that these are not OpenTelemetry general guidelines, but specific guidelines we adopt in this (js) repo.
But if you think it's better without I'll remove it